### PR TITLE
[PAY-1712] Implement purchases page

### PR DESCRIPTION
**Pull request by schottra**
### Description
This PR implements the Purchases pages.
* Added new `Table` derivative: `PurchasesTable`. It's very similar to the `AudioTransactionsTable`.
* Added `PurchasesPage` component, which fetches purchases and then renders a table
* Added a `getPurchases` to the User api
* Some small fixes to`useAllPaginatedQuery` (thanks @amendelsohn for the help!)
  * Don't attempt to denormalize data if no cache schema is provided, as it will convert arrays to objects and break logic in the pagination wrapper
  * Store the `status` separately and synchronize its updates with the data updates, so that we don't end up with a cycle where status is Success and the data is still empty. This was causing flashing of our empty state content in the table.
* Add a helper type to match the API response `USDSPurchaseDetails`
* Add xxxLarge icon variant to cover the 40x40 usage in the empty table content

### How Has This Been Tested?
Tested locally on Chrome, with flag on and off against my own test purchase data.

### Screenshots

NoPurchases
PurchasesLoading
Screenshot 2023-08-25 at 1 21 40 PM

### CODE REVIEW

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

File: packages/common/src/audius-query/hooks/usePaginatedQuery.ts [hunk 1]
```diff
@@ -80,7 +81,9 @@ export const useAllPaginatedQuery = <
       if (!statusIsNotFinalized(result.status)) {
         setLoadingMore(false)
       }
+      setStatus(result.status)
```
Comment (schottra) [inline packages/common/src/audius-query/hooks/usePaginatedQuery.ts:84]: cc @amendelsohn. This _seems_ like a safe change, but would appreciate your review here. It's not the perfect long term solution, but it does prevent the status getting set ahead of the data. 😅

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/pay-1712-purchases-table

**Comment by schottra** [comment]
@SocketSecurity ignore-all

File: packages/common/src/audius-query/hooks/usePaginatedQuery.ts (hunk 1 above)
Comment (amendelsohn) [reply packages/common/src/audius-query/hooks/usePaginatedQuery.ts:84]: Haha uhhhh yeah that seems fine ✅

File: packages/web/src/pages/purchases-page/PurchasesTable.tsx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    <div className={styles.contentName}>
+      <DynamicImage wrapperClassName={styles.artwork} image={image} />
+      <Text variant='body' size='small'>
+        {track.title}
+      </Text>
+    </div>
+  )
+}
+
+const UserNameAndBadge = ({ userId }: { userId: number }) => {
```
Comment (rickyrombo) [inline packages/web/src/pages/purchases-page/PurchasesTable.tsx:88]: nit: Badges (plural)

We should probably have a popover for the username

Also would this work? https://github.com/AudiusProject/audius-client/blob/62a0c72df7a0563e72f6db1b0bfe6ddc507735dd/packages/web/src/components/user-name-and-badges/UserNameAndBadges.tsx

**Review by rickyrombo — APPROVED** [review]
Awesome stuff, let's make some linear items for any follow ons (or update this PR if that's easier) and get you freed up for work on Previews!

File: packages/web/src/pages/purchases-page/PurchasesPage.tsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    useState<full.GetPurchasesSortDirectionEnum>(DEFAULT_SORT_DIRECTION)
+  const { mainContentRef } = useContext(MainContentContext)
+
+  const [count, setCount] = useState(0)
+
+  const {
+    status,
+    data: purchases
+    // hasMore,
+    // loadMore
```
[the remark this answers is not in the document]
Comment (schottra) [reply packages/web/src/pages/purchases-page/PurchasesPage.tsx:115]: These are uncommented and used in the next PR

File: packages/web/src/pages/purchases-page/PurchasesPage.tsx [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+      direction: PurchasesTableSortDirection
+    ) => {
+      setSortMethod(sortMethods[method] ?? DEFAULT_SORT_METHOD)
+      setSortDirection(sortDirections[direction] ?? DEFAULT_SORT_DIRECTION)
+    },
+    []
+  )
+
+  // TODO: Remove this short circuit once count is implemented
+  const fetchMore = useCallback(() => {}, [])
```
[the remark this answers is not in the document]
Comment (schottra) [reply packages/web/src/pages/purchases-page/PurchasesPage.tsx:142]: Yep, also used in the next PR

File: packages/web/src/pages/purchases-page/PurchasesPage.tsx [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+  // const fetchMore = useCallback(() => {
+  //   if (hasMore) {
+  //     loadMore()
+  //   }
+  // }, [hasMore, loadMore])
+
+  const onClickRow = useCallback((txDetails: USDCPurchaseDetails) => {
+    // TODO: Show details modal on row click
```
[the remark this answers is not in the document]
Comment (schottra) [reply packages/web/src/pages/purchases-page/PurchasesPage.tsx:150]: Already done, will add link in the next PR

**Pull request merged**